src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageControl.xaml.cs

```diff
@@ -70,7 +76,59 @@ private void ClearGridError(DataGrid dataGrid)
                 cellErrorInfo.SetValue(dataGrid, false, null);
                 rowErrorInfo.SetValue(dataGrid, false, null);
             }
-            catch (Exception) { }
+            catch (Exception) 
+            { 
+            }
+        }
+        
+        /// <summary>
+        /// Called when a property changes on the view model. Used to detect when the custom UI changes so that
+        /// the size of its grids can be determined
+        /// </summary>
+        private void ViewModel_PropertyChanged(Object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if(e.PropertyName.Equals(nameof(DebugPageViewModel.ActiveProviderUserControl)))
+            {
+                _customControlLayoutUpdateRequired = true;
```

Comment:
I'm not that familiar with XAML, so I'm assuming that these event handlers are always on the UI thread?